### PR TITLE
[core] remove opencensus/prometheus_exporter dependencies

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Depends on/includes #17181

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/agent.py [hunk 1]
```diff
@@ -13,8 +13,9 @@
 import aiohttp
 import aiohttp.web
 import aiohttp_cors
-import psutil
 from aiohttp import hdrs
+
+import psutil
```
Comment (simon-mo) [inline dashboard/agent.py:18]: move it to after import ray so it will use the our vendor version

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -2,8 +2,16 @@
 
 try:
     import aiohttp.web
+    import aiohttp
 except ImportError:
+    import aiohttp_cors  # noqa: F401
```
Comment (simon-mo) [inline dashboard/dashboard.py:7]: why are these imported in the except block?

File: python/requirements.txt [hunk 3]
```diff
@@ -10,7 +10,6 @@ aioredis
 click >= 7.0
 cloudpickle
 colorama
-colorful
```
Comment (simon-mo) [inline python/requirements.txt:13]: is there a separate PR for colorful?

**Review by simon-mo — APPROVED** [review]
Few small nits

File: python/requirements.txt (hunk 3 above)
Comment (richardliaw) [reply python/requirements.txt:13]: yep! we already did this a couple weeks ago.

**Pull request merged**